### PR TITLE
Update to elasticsearch 6

### PR DESCRIPTION
**Pull request by sonalkr132**
https://www.elastic.co/guide/en/elasticsearch/reference/6.8/removal-of-types.html
https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-version-migration.html

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,7 +25,10 @@ git:
   submodules: true
 
 before_install:
-  - git submodule update --init
+  - curl https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-6.8.13.deb -o elasticsearch.deb
```
Comment (colby-swandale) [inline .travis.yml:28]: Is there an apt mirror for elasticsearch v6 that we can use so we don't have to download and install everything manually?

File: Gemfile.lock [hunk 2]
```diff
@@ -133,19 +133,19 @@ GEM
     docile (1.3.2)
     domain_name (0.5.20190701)
       unf (>= 0.0.5, < 1.0.0)
-    elasticsearch (5.0.5)
-      elasticsearch-api (= 5.0.5)
-      elasticsearch-transport (= 5.0.5)
-    elasticsearch-api (5.0.5)
+    elasticsearch (7.5.0)
```
Comment (colby-swandale) [inline Gemfile.lock:136]: Does `elasticsearch` need to be restricted to `~> 6.0.0` also?

Comment (sonalkr132) [reply Gemfile.lock]: Thanks. It was fixed in 6.1, I have removed patch version pinning.

File: .travis.yml (hunk 1 above)
Comment (sonalkr132) [reply .travis.yml:28]: Travis needs to whitelist apt mirrors I think. The issue is following doesn't work on travis:
```
apt:
    sources:
      - elasticsearch-6.x
```
https://travis-ci.community/t/upgrade-elasticssearch-version-to-6-x-7-x/3423

We will need to install it manually. what we change here is dependence on artifacts.elastic.co by keeping the deb file in our s3.

**Pull request merged**